test/with_dummyserver/test_connectionpool.py

```diff
@@ -890,7 +896,7 @@ def test_preserves_path_dot_segments(self) -> None:
         """ConnectionPool preserves dot segments in the URI"""
         with HTTPConnectionPool(self.host, self.port) as pool:
             response = pool.request("GET", "/echo_uri/seg0/../seg2")
-            assert response.data == b"/echo_uri/seg0/../seg2"
+            assert response.data == b"/echo_uri/seg0/../seg2?"
```

Comment:
This is the same issue than in #3193 where Flask/Quart insists on adding a question mark.